contracts/drafts/voting/Voting.sol

```diff
@@ -0,0 +1,167 @@
+pragma solidity ^0.5.10;
+
+import "@hq20/fixidity/contracts/FixidityLib.sol";
+import "@openzeppelin/contracts/ownership/Ownable.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20Detailed.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../../state/StateMachine.sol";
+import "../../utils/SafeCast.sol";
+
+
+/**
+ * @title Issuance
+ * @dev Implements a simple voting process for proposals
+ *
+ * 1. Initialize the Voting with the votingToken address and the address and the callData of the proposal you wish to enact, should this Voting pass.
+ * 2. Setup the Voting by providing a quorum. The quorum must be expressed as an integer between 1 and 10000, representing a double digit percentage of the total supply of the voting tokens, with the comma shifted two digits to the right.
+ * 3. Open the voting.
+ * 4. Cast votes.
+ * 5. You can cancel your vote at any time. You should cancel it after the voting proposal has passed or after the voting has been cancelled.
+ * 6. Validate the quorum.
+ * 7. Enact the proposal.
+ */
+contract Voting is Ownable, StateMachine {
+    using SafeMath for uint256;
+    using FixidityLib for int256;
+    using SafeCast for int256;
+    using SafeCast for uint256;
+
+    event VotingCreated();
+    event QuorumSet();
+    event VoteCasted(address voter, uint256 votes);
+    event VoteCancelled(address voter, uint256 votes);
+
+    address public votingToken;
+    address public proposalContract;
+    bytes public proposalData;
+
+    address[] public voters;
+    mapping(address => uint256) public votes;
+
+    uint256 public quorum;
+
+    /**
+     * @dev Initialize the issuance with the token to issue and the token to
+     * accept as payment.
+     */
+    constructor(
+        address _votingToken,
+        address _proposalContract,
+        bytes memory _proposalData
```

Comment:
That's cool, I didn't know how to code the arguments of the proposal into a single variable :+1: